versioned_docs/version-v1.2/upgrade/v1-1-2-to-v1-2-0.md

```diff
@@ -398,3 +398,89 @@ If an upgrade is stuck in an `Upgrading System Service` state for an extended pe
   - https://github.com/harvester/harvester/issues/3863#issuecomment-1539681311
 
 ---
+### 8. An upgrade is stuck in the `Upgrading System Service` state due to the `customer provided SSL certificate without IP SAN` erro in `fleet-agent`
+
+If an upgrade is stuck in an `Upgrading System Service` state for an extended period, follow these steps:
+
+1. Find the upgrade related PODs:
+
+    ```
+    kubectl get pods -A | grep upgrade
+    ```
+
+    Example output:
+    ```
+    # kubectl get pods -A | grep upgrade
+    cattle-system               system-upgrade-controller-5685d568ff-tkvxb                 1/1     Running     0              85m
+    harvester-system            hvst-upgrade-vq4hl-apply-manifests-65vv8                   1/1     Running     0              87m  // waiting for managedchart to be ready
+    ..
+    ```
+
+2. The POD `hvst-upgrade-vq4hl-apply-manifests-65vv8` has following loop log:
+
+    ```
+    Current version: 102.0.0+up40.1.2, Current state: WaitApplied, Current generation: 23
+    Sleep for 5 seconds to retry
+    ```
+
+
+3. Check all bundle status, a couple of bundles are `OutOfSync`:
+
+    ```
+    # kubectl get bundle -A
+    NAMESPACE     NAME                                          BUNDLEDEPLOYMENTS-READY   STATUS
+    ...
+    fleet-local   mcc-local-managed-system-upgrade-controller   1/1
+    fleet-local   mcc-rancher-logging                           0/1                       OutOfSync(1) [Cluster fleet-local/local]
+    fleet-local   mcc-rancher-logging-crd                       0/1                       OutOfSync(1) [Cluster fleet-local/local]
+    fleet-local   mcc-rancher-monitoring                        0/1                       OutOfSync(1) [Cluster fleet-local/local]
+    fleet-local   mcc-rancher-monitoring-crd                    0/1                       WaitApplied(1) [Cluster fleet-local/local]
+    ```
+
+4. The POD `fleet-agent-*` has following error log:
+    ```
+    fleet-agent POD log:
+
+    time="2023-09-19T12:18:10Z" level=error msg="Failed to register agent: looking up secret cattle-fleet-local-system/fleet-agent-bootstrap: Post \"https://192.168.122.199/apis/fleet.cattle.io/    v1alpha1/namespaces/fleet-local/clusterregistrations\": tls: failed to verify certificate: x509: cannot validate certificate for 192.168.122.199 because it doesn't contain any IP SANs"
+    ```
```

Comment:
```suggestion
4. The pod `fleet-agent-*` has the following error log:

    ```
    fleet-agent POD log:

    time="2023-09-19T12:18:10Z" level=error msg="Failed to register agent: looking up secret cattle-fleet-local-system/fleet-agent-bootstrap: Post \"https://192.168.122.199/apis/fleet.cattle.io/    v1alpha1/namespaces/fleet-local/clusterregistrations\": tls: failed to verify certificate: x509: cannot validate certificate for 192.168.122.199 because it doesn't contain any IP SANs"
    ```
```